src/main/java/org/qualityannotate/coderepo/github/client/GithubApi.java

```diff
@@ -0,0 +1,98 @@
+package org.qualityannotate.coderepo.github.client;
+
+import io.quarkus.logging.Log;
+import org.apache.commons.lang3.tuple.Pair;
+import org.kohsuke.github.GHIssueComment;
+import org.kohsuke.github.GHPullRequest;
+import org.kohsuke.github.GHPullRequestReviewComment;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+import org.kohsuke.github.GitHubBuilder;
+import org.kohsuke.github.PagedIterator;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class GithubApi {
+    private final GitHubBuilder gitHubBuilder;
+    private final Map<Integer, GHPullRequest> pullRequestCache = new HashMap<>();
+    /**
+     * Guaranteed to be not null by {@link #init()}
+     */
+    private GitHub github = null;
+    /**
+     * Current user's id. Guaranteed to be not null by {@link #init()}
+     */
+    private long userId = 0L;
+
+    public GithubApi(String token) {
+        gitHubBuilder = new GitHubBuilder().withOAuthToken(token);
+    }
+
+    private void init() throws IOException {
+        if (github == null) {
+            github = gitHubBuilder.build();
+            github.checkApiUrlValidity();
+            userId = github.getMyself().getId();
+        }
+    }
+
+    private GHPullRequest getPR(String project, int pullRequestId) throws IOException {
+        init();
+        GHPullRequest ghPullRequest = pullRequestCache.get(pullRequestId);
+        if (ghPullRequest != null) {
+            return ghPullRequest;
+        }
+        GHRepository repository = github.getRepository(project);
+        ghPullRequest = repository.getPullRequest(pullRequestId);
+        pullRequestCache.put(pullRequestId, ghPullRequest);
+        return ghPullRequest;
+    }
+
+    public void createOrUpdateMainComment(String globalComment, String project, int pullRequestId) throws IOException {
+        init();
+        GHPullRequest pullRequest = getPR(project, pullRequestId);
+        PagedIterator<GHIssueComment> commentIterator = pullRequest.listComments().iterator();
+        while (commentIterator.hasNext()) {
+            GHIssueComment comment = commentIterator.next();
+            if (comment.getUser().getId() == userId) {
+                Log.info("Found comment: updating now");
+                comment.update(globalComment);
+                return;
+            }
+        }
+        System.out.println("Found comment: updating now");
+        pullRequest.comment(globalComment);
+    }
+
+    public void createOrUpdateFileComments(Map<Pair<String, Integer>, String> fileLineToComment, String project,
```

Comment:
🛑 maintainability: Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [details](https://sonarcloud.io/organizations/quyt/rules?open=java:S3776&rule_key=java:S3776)